smdebug/tensorflow/keras.py

```diff
@@ -550,12 +550,28 @@ def on_test_batch_begin(self, batch, logs=None):
     def on_predict_batch_begin(self, batch, logs=None):
         self._on_any_batch_begin(batch, ModeKeys.PREDICT, logs=logs)
 
+    def _write_optimizer_variables(self):
+        optimizer_collections = self.collection_manager.get(CollectionKeys.OPTIMIZER_VARIABLES)
+        if optimizer_collections in self._get_collections_to_save_for_step():
```

Comment:
What if user provide regex for optimizer variables? Will it still be saved?
Can you please Try an example with : regex of optimizer variables 
and
A custom collection with regex of optimizer variables.

Reply:
I have added a test which creates custom collection with regex of optimizer variables.

`hook.get_collection("custom_optimizer_variables").include("Adam")`